### DISCUSSION TITLE
How to get a score within 0 and 1?

### DISCUSSION THREAD

**Discussion by KunmingS** [milvus-io/milvus#19921] [Q&A and General discussion] [answered]
I am suing IVF_FLAT, and metric_type is L2. When I conduct a vector similarity search, I got same value of hit.distance and hit.score, both higher than 100. I also tried IP metric_type, but it still much higher than 1. I saw many people got a score between 0 and 1, how could I got this value? I want to use this value to judge the similarity between the vector to search and the results. Thanks!

**Comment by bingbb47** [comment, marked as answer]
may you have to normalized vectors

**Reply by KunmingS** [reply]
> may you have to normalized vectors

Thank you for your reply!

I have normalized vectors before insert into MILVUS. I use the result of FC layer of resnet 50. They are all between 0 and 1. But the distance of searching result and score are so big. Too strange

**Reply by yhmo** [reply]
Milvus supports two kinds of metric types: IP and L2
For IP metric, make sure all the vectors are normalized, including the target vectors that you input to search() interface.
Distance of IP metric always between 0 and 1. If distance is 1.0, the two vectors are equal.

Example:
```
search_param = {
        "data": normalized_vectors,
        "anns_field": vector_field_name,
        "param": {"metric_type":"IP", "params": {}},
        "limit": 10}
results = collection.search(**search_param)
for hits in results:
    for hit in hits:
        print(f"hit: {hit}")
```

**Reply by KunmingS** [reply]
> Milvus supports two kinds of metric types: IP and L2 For IP metric, make sure all the vectors are normalized, including the target vectors that you input to search() interface. Distance of IP metric always between 0 and 1. If distance is 1.0, the two vectors are equal.
> 
> Example:
> 
> ```
> search_param = {
>         "data": normalized_vectors,
>         "anns_field": vector_field_name,
>         "param": {"metric_type":"IP", "params": {}},
>         "limit": 10}
> results = collection.search(**search_param)
> for hits in results:
>     for hit in hits:
>         print(f"hit: {hit}")
> ```

Thanks for your suggestion! I found that I did not normalize the vector to search. This solved my problem.
